### DISCUSSION TITLE
showBuildSettings&resolvePackageDependencies eat 20s of build time

### DISCUSSION THREAD

**Discussion by yakimant** [fastlane/fastlane#17563] [General] [answered]
It's not a problem, until you have 70+ test schemes to run on CI.
It become 23m+.

Do we really need to run those? What is the reason?

FYI
https://github.com/fastlane/fastlane/issues/14777
https://github.com/fastlane/fastlane/issues/15717

**Comment by rogerluan** [comment]
Very interesting.

Digging into the links you posted I found this answer as to some of the "why's": https://github.com/fastlane/fastlane/pull/15732#discussion_r351858752

Does this answer your question? 🤔

**Comment by ethan-gates** [comment, marked as answer]
I've also been running into these over the past year or so of fastlane releases.  For me, I was running into this while running scan, and looking at the fastlane core project model, it looks like fastlane will call `xcodebuild -showBuildSettings` when it needs a value that it doesn't have (via normal lane arguments).  In particular, for my case I learned the absence of 'app_name', 'device', and 'deployment_target_version' arguments as well as the presence of the 'destination' argument can all implicitly trigger the show build settings command.  Earlier this year I added those arguments (removing 'destination') and it removed the call for a few releases.

Just today I looked into why fastlane started running the command again despite keeping scan arguments the same, and learned that it's now checking for 'SUPPORTS_MACCATALYST' in the project file (as of a few releases ago).  

It's a little difficult to determine what lane arguments can suffice for which build settings since AFAIK that mapping doesn't exist nicely in fastlane, but you can get an idea by looking at which build settings fastlane is looking for.

You can trace what build setting triggered the show build settings command by looking at the key argument to [this function in project.rb](https://github.com/fastlane/fastlane/blob/4a024011bd9ad788346d256d23e2ce28b0189101/fastlane_core/lib/fastlane_core/project.rb#L361) (using fastlane as a local gem)

**Reply by yakimant** [reply]
Thanks for your information, @ethantakeshi!

Sounds tricky to find, what can cause to add 20s.
We are currently investigating a switch from scan to xcodebuild Fastalne action, which doesn't have this issue.
